core/src/Kokkos_Core_fwd.hpp

```diff
@@ -325,6 +327,91 @@ struct LAnd;
 template <class ScalarType, class Space = HostSpace>
 struct LOr;
 
+//--------------------------------------------------------------------------------------//
+//  this is used to mark execution or memory spaces as not available
+//
+template <typename Tp>
```

Comment:
Just curious (not a request to change):  why default to available and explicitly mark those that are unavailable?

Also, based on its name it looks like it could be applied to any type (not just execution and memory spaces).  Would `IsSpaceAvailable` be a better name?

Reply:
I just took this from @jrmadsen's pull requests and it's currently unused. I'm open to remove this and add something later if desired.

Reply:
@nliber I have used that style in timemory bc it ensures that someone cannot specialize something to be turned on that there isn't support for in the build system, i.e. explicit specialization on the types that definitely should be turned off. Although this is probably less of a concern in Kokkos than it is in timemory.